### PR TITLE
Adding Go-fmt tools and fixing go-fmt errors

### PR DESCRIPTION
**Pull request by poothia**
part of #302

### CODE REVIEW

**Comment by irfanurrehman** [comment]
@poothia thanks for doing this. There is a basic `fmt` target written in root Makefile, but it does not exhaustively include all project files. It would be a good idea to integrate `hack/update-gofmt.sh` into that target. 
Makefile has some additional targets, for example `make vet`, invoking `go vet`, which might be useful as a CI check.
We have been intending to update the CI checks invoking make rather then calling various scripts.
If that might interest you, please feel free to create an issue and claim it.

File: hack/update-gofmt.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eou pipefail
+
+SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+REPO_ROOT="$(cd "${SCRIPT_ROOT}/../.." && pwd)"
+pushd ${REPO_ROOT} > /dev/null
+
+find . -name "*.go" | grep -v -e "\/vendor\/" -e "/*deepcopy.go" -e "/*kubebuilder.go" -e "/doc.go" | xargs gofmt -s -w
```
Comment (irfanurrehman) [inline hack/update-gofmt.sh]: Why exclude `doc.go` ?

File: hack/go-tools/verify-gofmt.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+REPO_ROOT="$(cd "${SCRIPT_ROOT}/../.." && pwd)"
+pushd ${REPO_ROOT} > /dev/null
+
+find_files() {
+  find . -not \( \
+      \( \
+        -wholename '*/vendor/*' \
+        -o -wholename '*/*kubebuilder.go' \
+        -o -wholename '*/*deepcopy.go' \
+        -o -wholename '*/doc.go' \
```
Comment (irfanurrehman) [inline hack/go-tools/verify-gofmt.sh]: Why exclude `doc.go`?

Comment (gyliu513) [reply hack/go-tools/verify-gofmt.sh]: @irfanurrehman please refer to some discussion here https://github.com/kubernetes-sigs/federation-v2/pull/304#pullrequestreview-168197489

**Comment by poothia** [comment]
@irfanurrehman @gyliu513 thanks for reviewing this.
@irfanurrehman i have created an issue #364 regarding CI and makefile integration. i'll investigate and try to fix this.

**Comment by marun** [comment]
@poothia Thank you for doing this.  In future, please separate code additions from mechanical changes in separate commits.

/lgtm
/approve

**Pull request merged**